src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String datum : data) {
+                String[] parts = datum.split(" ");
+                int hours = Integer.parseInt(parts[2]);
+                int income = Integer.parseInt(parts[3]);
+                if (parts[1].equals(name) && dateComparator(parts[0], dateFrom, dateTo)) {
+                    salary += hours * income;
+                }
+            }
+            result.append(System.lineSeparator()).append(name).append(" - ").append(salary);
+        }
+
+        return new String(result);
```

Comment:
![image](https://github.com/user-attachments/assets/b3e9883b-7012-483b-8bda-93e6b895f506)

Reply:
How i can return a stringBuilder in "public string" class if i didn' change it into string?

Reply:
As AI Mentor said, use `.toString()` method on the stringBuilder.